collated/main/A0138411N.md

```diff
@@ -0,0 +1,877 @@
+                    
+                if (t1.getStartDateTime().equals(t2.getStartDateTime())
+                    && (t1.getEndDateTime().equals(t2.getEndDateTime()))) {
+                    return t1.getTaskName().taskName.compareTo(t2.getTaskName().taskName);
+                } 
+                
+                if (t1.getStartDateTime().before(t2.getStartDateTime())) {
+                    return -1;
+                } else if (t1.getStartDateTime().after(t2.getStartDateTime())) {
+                    return 1;
+                } 
+                
+                if (t1.getEndDateTime().before(t2.getEndDateTime())) {
+                    return -1;
+                } else if (t1.getEndDateTime().after(t2.getEndDateTime())) {
+                    return 1;
+                }
+                
+                return 0;
+            }
+        };
+        return new UnmodifiableObservableList<>(new SortedList(filteredTasks, compareDateTime));
+    }
+```
+###### /main/java/seedu/ggist/model/ModelManager.java
+``` java
+    @Override
+    public void updateFilteredListToShowDate(String keywords){
+        updateFilteredTaskList(new PredicateExpression(new DateQualifier(keywords)));
+    }
+
+    private void updateFilteredListToShowDate(Expression expression) {
+        filteredTasks.setPredicate(expression::satisfies);
+    }
+```
+###### /main/java/seedu/ggist/model/ModelManager.java
+``` java
+    private class DateQualifier implements Qualifier {
+        private String taskDateKeyWords;
+
+        DateQualifier(String taskDateKeyWords) {
+            this.taskDateKeyWords = taskDateKeyWords;
+        }
+
+        @Override
+        public boolean run(ReadOnlyTask task) {
+            if (taskDateKeyWords == null) {
+                return true;
+            }
+            return ((taskDateKeyWords.equalsIgnoreCase(task.getStartDate().toString()) || 
+                   taskDateKeyWords.equalsIgnoreCase(task.getEndDate().toString())) && !task.isDone()) ||
+                   ((task.getStartDate().value.equals(Messages.MESSAGE_NO_START_DATE_SPECIFIED) && 
+                   (task.getStartTime().value.equals(Messages.MESSAGE_NO_START_TIME_SET) &&
+                   (task.getEndDate().value.equals(Messages.MESSAGE_NO_END_DATE_SPECIFIED) && 
+                   (task.getEndTime().value.equals(Messages.MESSAGE_NO_END_TIME_SET)) && !task.isDone())) ||
+                   (task.isOverdue() && !task.isDone())));
+        }
+```
+###### /main/java/seedu/ggist/model/task/DeadlineTask.java
+``` java
+    /**
+     * Every field must be present and not null.
+     * @throws IllegalValueException 
+     */
+    public DeadlineTask(TaskName taskName, TaskDate taskDate, TaskTime startTime, TaskDate endDate, TaskTime endTime, Priority priority) throws IllegalValueException {
+        super(taskName, taskDate, startTime, endDate, endTime, priority);
+    }
+    
+    /**
+     * Copy constructor.
+     * @throws IllegalValueException 
```

Comment:
either include why it throws the exception or remove the line; as such it doesn't give any additional information in the current form.